app/code/Magento/Sales/Model/Order.php

```diff
@@ -310,7 +318,6 @@ public function __construct(
         \Magento\Catalog\Model\Product\Visibility $productVisibility,
         \Magento\Sales\Api\InvoiceManagementInterface $invoiceManagement,
         \Magento\Directory\Model\CurrencyFactory $currencyFactory,
-        \Magento\Eav\Model\Config $eavConfig,
```

Comment:
you cannot remove the injection via the `__constuct` without this breaking BC but you can remove the setting of `$this->_eavConfig = $eavConfig;` which you have done. If you add back this to the construct and docblock the rest seems fine to me